doc/storagemond/storagemond-hld.md

```diff
@@ -0,0 +1,522 @@
+key                        = STORMOND_CONFIG|INTERVALS   ; This key is for information about stormon daemon polling interval configurations
+
+; field                    = value
+
+daemon_polling_interval    = UINT32               ; The polling frequency for reading dynamic information
+fsstats_sync_interval      = UINT32               ; The frequency of FSIO Reads/Writes synchronization to location on disk
+```
+
+Example: `stormond` configured with daemon polling interval of 60s and the JSON file sync interval as 360 seconds:
+
+```
+127.0.0.1:6379[4]> KEYS STORMOND*
+1) "STORMOND_CONFIG|INTERVALS"
+127.0.0.1:6379[4]> HGETALL "STORMOND_CONFIG|INTERVALS"
+1) "daemon_polling_interval"
+2) "60"
+3) "fsstats_sync_interval"
+4) "360"
+127.0.0.1:6379[4]>
+
+```
+
+**YANG Model**
+
+```
+container sonic-stormond-config {
+
+        container STORMOND_CONFIG {
+
+            description "stormond_config table in config_db.json";
+
+            container INTERVALS {
+
+                leaf daemon_polling_interval {
+                    description "Polling inerval for Storage Monitoring Daemon in STORMOND_CONFIG table";
+                    type uint32 {
+                        range "1..4294967295";
+                    }
+                    default "3600";
+                }
+
+                leaf fsstats_sync_interval {
+                    description "FSSTATS JSON file syncing interval for the Storage Monitoring Daemon in STORMOND_CONFIG table";
+                    type uint32 {
+                        range "1..4294967295";
+                    }
+                    default "86400";
+                }
+            }
+        }
+    }
+```
+
+## **4. Test Plan**
+
+The following is the `sonic-mgmt` test plan for the daemon. The first column represents various testing scenarios. The other columns represent the intended status of various components of the daemon.
+
+| **Event**              | **State_DB** | **JSON**  |  **PROCFS STATUS**      | **JSON SYNCED WITH `STATE_DB`?** | **STORMON RESTARTED** |
+| ---------------------- | ------------ | --------- | ----------------------- | ----------------------------- | --------------------- |
+|                        |              |           |                         |                               |                       |
+| Init                   | CLEARED      | CLEARED   | CLEARED, Initial Values | YES                           | YES                   |
+| Planned Cold Reboot    | CLEARED      | NOT RESET | CLEARED, Initial Values | YES                           | YES                   |
+| Planned Soft Reboot    | CLEARED      | NOT RESET | CLEARED, Initial Values | YES                           | YES                   |
+| Planned Warm Reboot    | NOT RESET    | NOT RESET | CLEARED, Initial Values | YES                           | YES                   |
+| Unplanned daemon crash | NOT RESET    | NOT RESET | NOT RESET               | UNSURE                        | YES                   |
```

Comment:
@assrinivasan RESET means?

Reply:
RESET is supposed to signify the values persist. Will change "NOT RESET" to "PERSISTED".